go/test/endtoend/vreplication/cluster_test.go

```diff
@@ -537,9 +538,27 @@ func (vc *VitessCluster) AddShards(t *testing.T, cells []*Cell, keyspace *Keyspa
 		require.NotEqual(t, 0, primaryTabletUID, "Should have created a primary tablet")
 		log.Infof("InitializeShard and make %d primary", primaryTabletUID)
 		require.NoError(t, vc.VtctlClient.InitializeShard(keyspace.Name, shardName, cells[0].Name, primaryTabletUID))
+
 		log.Infof("Finished creating shard %s", shard.Name)
 	}
 
+	log.Infof("Applying throttler config for keyspace %s", keyspace.Name)
+	_, err := throttler.UpdateThrottlerTopoConfigRaw(vc.VtctldClient, keyspace.Name, true, false, 30*time.Second.Seconds(), "")
+	require.NoError(t, err)
+	log.Infof("Waiting for throttler config to be applied on all shards")
+	for _, shard := range keyspace.Shards {
+		for _, tablet := range shard.Tablets {
+			clusterTablet := &cluster.Vttablet{
+				Alias:    tablet.Name,
+				HTTPPort: tablet.Vttablet.Port,
+			}
+			log.Infof("+ Waiting for throttler config to be applied on %s, type=%v", tablet.Name, tablet.Vttablet.TabletType)
+			throttler.WaitForThrottlerStatusEnabled(t, clusterTablet, true, nil, time.Minute)
```

Comment:
The merchant keyspace gets `Reshard`ed many times. When it's first created it has the `-80, 80-` shards. It e.g. then gets resharded to `-40, 40-c0, c0-`. I don't see where we wait for the throttler to actually be enabled on these new tablets/shards though. This may be part of the problem.